specificatie/gezag.yaml

```diff
@@ -0,0 +1,101 @@
+openapi: 3.0.3
+info:
+  title: Gezag definities
+  version: 2.0.0
+  contact: {}
+paths: {}
+components:
+  schemas:
+    GezagPersoonBeperkt:
+      allOf:
+        - $ref: 'persoon.yaml#/components/schemas/PersoonBeperkt'
+        - type: object
+          properties:
+            gezag:
+              $ref: "#/components/schemas/Gezag"
+    Gezag:
+      type: array
+      items: 
+        $ref: '#/components/schemas/AbstractGezagsrelatie'
+      minItems: 0
+    Meerderjarige:
+      type: object
+      properties:
+        burgerservicenummer:
+          $ref: 'persoon.yaml#/components/schemas/Burgerservicenummer'
+    Minderjarige:
+      type: object
+      properties:
+        burgerservicenummer:
+          $ref: 'persoon.yaml#/components/schemas/Burgerservicenummer'
+      required: 
+        - burgerservicenummer
+    Ouder:
+      allOf: 
+        - $ref: '#/components/schemas/Meerderjarige'
+      required: 
+        -  burgerservicenummer
+    AbstractGezagsrelatie:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          type: string
+      discriminator:
+        propertyName: type
+        mapping:
+          TweehoofdigOuderlijkGezag: '#/components/schemas/TweehoofdigOuderlijkGezag'
+          EenhoofdigOuderlijkGezag: '#/components/schemas/EenhoofdigOuderlijkGezag'
+          GezamenlijkGezag: '#/components/schemas/GezamenlijkGezag'
+          Voogdij: '#/components/schemas/Voogdij'
+          GezagNietTeBepalen: '#/components/schemas/GezagNietTeBepalen'
+    TweehoofdigOuderlijkGezag:
+      allOf:
+        - $ref: '#/components/schemas/AbstractGezagsrelatie'
+        - type: object
+          properties:
+            ouders:
+              type: array
+              items:
+                $ref: '#/components/schemas/Ouder'
+              minItems: 2
+              maxItems: 2
+            minderjarige:
+              $ref: '#/components/schemas/Minderjarige'
+    EenhoofdigOuderlijkGezag:
+      allOf:
+        - $ref: '#/components/schemas/AbstractGezagsrelatie'
+        - type: object
+          properties:
+            ouder:
+              $ref: '#/components/schemas/Ouder'
+            minderjarige:
+              $ref: '#/components/schemas/Minderjarige'
+    GezamenlijkGezag:
+      allOf:
+        - $ref: '#/components/schemas/AbstractGezagsrelatie'
+        - type: object
+          properties:
+            ouder:
+              $ref: '#/components/schemas/Ouder'
+            derde:
+              $ref: '#/components/schemas/Meerderjarige'
+            minderjarige:
+              $ref: '#/components/schemas/Minderjarige'
+    Voogdij:
+      allOf:
+        - $ref: '#/components/schemas/AbstractGezagsrelatie'
+        - type: object
+          properties:
+            derden:
+              type: array
+              items:
+                $ref: '#/components/schemas/Meerderjarige'
+              minItems: 1
```

Comment:
wanneer er voogdij is vanuit een gerechtelijke uitspraak is niet bekend wie de voogd is. Dus minItems zou m.i. 0 moeten zijn

Reply:
aangepast